lib/src/lexer.c

```diff
@@ -310,6 +310,7 @@ void ts_lexer_init(Lexer *self) {
     .included_range_count = 0,
     .current_included_range_index = 0,
   };
+  self->data.line = &self->current_position.extent.row;
```

Comment:
It's not a good idea to share a pointer every time, it's not better than to make a copy of the value directly. Tree-sitter's scanner API provides methods for most cases to be maximally lazy.

Reply:
Removed